pkg/engine/context/loaders/globalcontext.go

```diff
@@ -75,15 +75,10 @@ func (g *gctxLoader) loadGctxData() ([]byte, error) {
 		g.logger.Error(err, "context entry does not have resource cache")
 		return nil, fmt.Errorf("resource cache not found")
 	}
-	rc, err := variables.SubstituteAllInType(g.logger, g.enginectx, g.entry.GlobalReference)
```

Comment:
Why remove this? The user could have template variables in the name. Maybe unlikely, but this would be a breaking change.

Reply:
```
	rc, err := variables.SubstituteAllInType(g.logger, g.enginectx, g.entry.GlobalReference)
	if err != nil {
		return nil, err
	}
	g.logger.V(6).Info("variables substituted", "resourcecache", rc)
```
```
	path, err := variables.SubstituteAll(g.logger, g.enginectx, rc.JMESPath)
	if err != nil {
		g.logger.Error(err, "failed to substitute variables in context entry")
		return nil, fmt.Errorf("failed to substitute variables in context entry %s JMESPath %s: %w", g.entry.Name, rc.JMESPath, err)
	}
```
I think these two pieces of code are duplicated. I didn't consider that name might use template variables.

Reply:
keep [L85-L89](https://github.com/kyverno/kyverno/blob/v1.13.4/pkg/engine/context/loaders/globalcontext.go#L85-L89) and remove [L124-L128](https://github.com/kyverno/kyverno/blob/v1.13.4/pkg/engine/context/loaders/globalcontext.go#L124-L128)